native/src/modules/common/components/ProgressBar.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import * as React from 'react'
+import Svg, { Circle, Image } from 'react-native-svg'
+import { Dimensions } from 'react-native'
+import styled from 'styled-components/native'
+import { type StyledComponent } from 'styled-components'
+import { buildConfigAssets } from '../../app/constants/buildConfig'
+import type { ThemeType } from '../../theme/constants'
+import { type TFunction } from 'react-i18next'
+
+const Container: StyledComponent<{||}, {||}, *> = styled.View`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`
+
+const Text = styled.Text`
+  padding-top: 25px;
+  font-size: 20px;
+  font-weight: 700;
+`
+
+const { width } = Dimensions.get('window')
+const size = width / 3 // eslint-disable-line no-magic-numbers
+
+const logoSize = size * 0.6 // eslint-disable-line no-magic-numbers
+const logoXY = (size - logoSize) / 2
+const logoToCenter = `translate(${logoXY}, ${logoXY})`
+
+// progress
+const strokeWidth = size * 0.09 // eslint-disable-line no-magic-numbers
+const r = (size - strokeWidth) / 2
+const cx = size / 2
+const cy = size / 2
+const circumference = r * 2 * Math.PI
+const strokeDasharray = circumference
+
+type PropsType = {|
+  progress: number,
+  theme: ThemeType,
+  t: TFunction
+|}
+
+class ProgressBar extends React.Component<PropsType> {
+  render () {
+    const { t, progress, theme } = this.props
+    const strokeDashoffset = circumference - progress * circumference
```

Comment:
I think this is already handled: How many times does progress update per loading process? Is this bound to an upper limit?

I think yes, just checking :)

Reply:
The progress event only fires in 10-percent-step-width (0.10, 0.20, .. 0.9), so it only updates 10 times